src/components/JobsContext.jsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import { JOBS_DOCTYPE } from '../doctypes'
+import CozyRealtime from 'cozy-realtime'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import logger from 'cozy-logger'
+import { useI18n } from 'cozy-ui/react/I18n'
+
+const log = logger.namespace('import.context')
+
+export const JobsContext = createContext({})
+
+export const useJobsContext = () => {
+  return useContext(JobsContext)
+}
+
+const JobsProvider = ({ children, client }) => {
+  const [jobsInProgress, setJobsInProgress] = useState([])
+  const { t } = useI18n()
+  const realtimeStartedRef = useRef(false)
+  const jobsInProgressRef = useRef([])
+
+  const realtime = useMemo(() => {
+    return new CozyRealtime({ client })
+  }, [client])
+
+  const handleRealtime = data => {
+    const { worker, state, message: msg } = data
+    const { current: currJobsInProgress } = jobsInProgressRef
+    const index = currJobsInProgress.findIndex(a => a.account === msg.account)
+    const exist = index !== -1
+
+    let arr = [...currJobsInProgress]
+    if (worker === 'konnector') {
+      if (state === 'running' && !exist) {
+        arr.push(msg)
+      } else if (state === 'done' && exist) {
+        arr.splice(index, 1)
+        Alerter.success(t('JobsContext.alerter-success'))
+      } else if (state === 'errored' && exist) {
+        arr.splice(index, 1)
+        Alerter.error(t('JobsContext.alerter-errored'))
+      }
+      jobsInProgressRef.current = arr
+      setJobsInProgress(jobsInProgressRef.current)
+    }
+  }
+
+  const startJobsRealtime = () => {
+    if (!realtimeStartedRef.current) {
+      log('info', 'Start Jobs Realtime')
+      realtimeStartedRef.current = true
+
+      realtime.subscribe('created', JOBS_DOCTYPE, handleRealtime)
+      realtime.subscribe('updated', JOBS_DOCTYPE, handleRealtime)
+      realtime.subscribe('deleted', JOBS_DOCTYPE, handleRealtime)
+    }
+  }
+
+  const stopJobsRealtime = () => {
+    if (realtimeStartedRef.current) {
+      log('info', 'Stop Jobs Realtime')
+      realtimeStartedRef.current = false
+
+      realtime.unsubscribe('created', JOBS_DOCTYPE, handleRealtime)
+      realtime.unsubscribe('updated', JOBS_DOCTYPE, handleRealtime)
+      realtime.unsubscribe('deleted', JOBS_DOCTYPE, handleRealtime)
+    }
+  }
+
+  useEffect(() => {
+    startJobsRealtime()
+    return () => stopJobsRealtime()
+  }, [])
+
+  return (
+    <JobsContext.Provider
+      value={{
+        jobsInProgress
```

Comment:
Nit: It could be nice to prepare a bit the context value to make its consumption easier for the components below. For example, indexing the accounts or the triggers, or the connectors.

```
  const value = useMemo(() => ({
    inProgress: jobsInProgress,
    byAccountId: keyBy(jobsInProgress, x => x.account),
    byKonnectorSlug: keyBy(jobsInProgress, x => x.konnector),,
    byTriggerId: keyBy(jobsInProgress, x => x.trigger), // useMemo could be used 
  }, [jobsInProgress])
  return <JobsContext.Provider
    value={value}
  ...
  />
```

The value passed below is here memoized not to incure unnecessary renders (not sure if this is really necessary but it cannot hurt I guess) and indexed so that a component can access info easily.

Reply:
Yes I think so too, it would be better in realtime lib